test/errors/worker-error.js

```diff
@@ -0,0 +1,65 @@
+'use strict'
+
+const Lab = require('lab')
+const lab = exports.lab = Lab.script()
+const describe = lab.describe
+const beforeEach = lab.beforeEach
+const afterEach = lab.afterEach
+const it = lab.test
+const expect = require('code').expect
+const sinon = require('sinon')
+
+const WorkerError = require('../../errors/worker-error')
+
+describe('errors', () => {
+  describe('WorkerError', () => {
+    describe('constructor', () => {
+      beforeEach((done) => {
+        sinon.spy(WorkerError.prototype, 'setQueue')
+        sinon.spy(WorkerError.prototype, 'setJob')
+        done()
+      })
+
+      afterEach((done) => {
+        WorkerError.prototype.setQueue.restore()
+        WorkerError.prototype.setJob.restore()
+        done()
+      })
+
+      it('should set the queue name', (done) => {
+        const queue = 'neat:stuff'
+        const error = new WorkerError('', {}, {}, queue)
+        expect(WorkerError.prototype.setQueue.calledWith(queue)).to.be.true()
+        expect(error.data.queue).to.equal(queue)
+        done()
+      })
+
+      it('should set the job', (done) => {
+        const job = { type: 'neat:stuff' }
+        const error = new WorkerError('', {}, {}, '', job)
```

Comment:
The api would be way nicer if you could just do `new WorkerError('awesome', job)` or even `new WorkerError('awesome', { job: job })`